F 07

vladn255 opened the pull request.

vladn255 committed:
F-07. Completed

- minor location-form refactoring;
- implemented posting order;
- implemented checkout server loading

vladn255 committed:
F-07. bugfix

- minor refactoring at features-form, model-photo;
- fixing map behaviour: switching to active marker, resetting pickpoints when new city is choosed

AlexsandrLitkevich commented on src/components/location-form/location-form.jsx:
```diff
@@ -74,15 +69,29 @@ const LocationForm = () => {
 
             if (filteredCity.length) {
-                setActiveCity(filteredCity[0].value)
+                setActiveCity({
```
Почему тут нулевой индекс?Их может быть больше?

AlexsandrLitkevich commented on src/components/location-form/location-form.jsx:
```diff
@@ -74,15 +69,29 @@ const LocationForm = () => {
+                    name: filteredCity[0].value
+                })
+
+                const filteredPoints = pickpointData.filter((point) => point.cityId !== null && point.cityId.name === filteredCity[0].value)
```
point.cityId  && point.cityId.name === filteredCity[0].value так лучше читается

AlexsandrLitkevich commented on src/components/order-receipt/order-receipt.jsx:
```diff
@@ -135,7 +140,7 @@ const OrderReceipt = () => {
             <ul className="receipt__list">
-                {details.filter((detail) => detail.value.length !== 0)
+                {details.filter((detail) => detail.value)
                     .map((detail) => <OrderReceiptItem key={detail.name} name={detail.name} value={detail.text} />)}
```
name уникальное и постоянное значение?Подходит для key?

AlexsandrLitkevich commented on src/store/api-action.js:
```diff
@@ -44,10 +50,36 @@ const fetchRateDataEntity = () => (_dispatch, _getState, api) => (
+        .catch((err) => console.log(err))
+)
+
+const deleteOrder = (orderId) => (_dispatch, _getState, api) => (
```
Что это делает?почему они консолы?

vladn255 committed:
F-07. Bugfixes_2

- location-form refactoring;
- excluding delete fetch;
- order-receipt refactoring;

vladn255 committed:
minor fixes

vladn255 committed:
F-07. Bugfixes

- date pick reworked using "react-datetime-picker";
- fixed scaling;
- changed logo title at header into a link;
- refactoring at location-form (waiting for pickpoints coordinates loading)
